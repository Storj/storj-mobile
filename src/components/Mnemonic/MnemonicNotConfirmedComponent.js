import {
    View,
    Text,
    Image,
    TouchableOpacity,
    Alert,
    StyleSheet
} from 'react-native';
import React, { Component } from 'react';
import mnemonicScreenConstants from '../../utils/constants/mnemonicScreenConstants';
import { getWidth, getHeight } from '../../utils/adaptive';

export default class MnemonicNotConfirmedComponent extends Component {
    constructor(props) {
        super(props);
    }
    
    render() {
        return(
            <View style = { styles.mainContainer }>
                <View style = { styles.contentContainer }>
                    <View style = { styles.titleContainer } >
                        <Text style = { styles.titleText } >Secret phrase backup failed</Text>
                    </View>
                    <View style = { styles.errorImageContainer }>
                        <Image
                            source = { require('../../images/RegisterInfoScreens/Error.png') }
                            style = { styles.errorImage }
                            resizeMode = 'contain' />
                    </View>
                    <TouchableOpacity onPress = { () => { 
                        this.props.navigation.state.params.setNewData(); 
                        this.props.screenProps.navigateBack(); } }>
                        
                        <View style = { styles.backButton } >
                            <Text style = { styles.backButtonText }>Back</Text>
                        </View>
                    </TouchableOpacity>
                    <TouchableOpacity onPress = { () => { this.props.screenProps.redirectToLoginScreen(); } } >
                        <View style = { styles.nextButton }>
                            <Text style = { styles.nextButtonText }>Skip</Text>
                        </View>
                    </TouchableOpacity>
                </View>
            </View>
        );
    }   
}

const styles = StyleSheet.create({
    mainContainer: { 
        flex: 1, 
        backgroundColor: '#FFFFFF'
    },
    contentContainer: {
        paddingHorizontal: getWidth(20)
    },
    titleContainer: { 
        marginTop: getHeight(30),
        height: getHeight(70),
        flexDirection: 'row' 
    },
    titleText: { 
        fontFamily: 'Montserrat-Bold', 
        fontSize: getHeight(27), 
        lineHeight: getHeight(33),
        color: '#384B65',
        marginLeft: getWidth(15)
    },
    titleTextContainer: {
        width: getWidth(191)
    },
    errorImage: {
        height: getHeight(250),
        width: getWidth(375)
    },
    errorImageContainer: {
        height: getHeight(250),
        marginTop: getHeight(75),
        alignItems: 'center',
        justifyContent: 'center'
    },
    infoContainer: {
        marginTop: getHeight(45)
    },
    infoText:{
        fontFamily: 'Montserrat-Regular', 
        fontSize: getHeight(16), 
        lineHeight: getHeight(23),
        color: '#384B65'
    },
    backButton: {
        marginTop: getHeight(60),
        alignSelf: 'center',
        width: getWidth(335),
        height: getHeight(50),
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: '#2782ff',
        borderColor: '#2794FF',
        borderRadius: getWidth(6),
        borderWidth: getWidth(1.5)
    },
    backButtonText: {
        fontFamily: 'Montserrat-Bold',
        fontSize: getHeight(14),
        color: 'white'
    },
    nextButton: { 
        marginTop: getHeight(20),
        width: getWidth(335),
        height: getHeight(50),
        borderRadius: 6,
        borderWidth: getWidth(1.5),
        borderColor: '#2794FF',
        justifyContent: 'center',
        alignItems: 'center'
    },
    nextButtonText: { 
        fontFamily: 'Montserrat-SemiBold', 
        fontSize: getHeight(16), 
        color: '#2794FF' 
    },
});