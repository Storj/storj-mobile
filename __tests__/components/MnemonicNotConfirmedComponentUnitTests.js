import {
    View
} from 'react-native';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import sinon from 'sinon';
import Adapter from 'enzyme-adapter-react-16';
import MnemonicNotConfirmedComponent from '../../src/components/Mnemonic/MnemonicNotConfirmedComponent';

Enzyme.configure({ adapter: new Adapter() });

describe('MnemonicNotConfirmedComponent', () => {    

	it('renders correctly', () => {

		const wrapper = shallow(
			<MnemonicNotConfirmedComponent />
        );
        
		expect(wrapper).toMatchSnapshot();
    });

    it('triggers actions correctly on TouchableOpacity press', () => {

        const redirectToLoginScreenSpy = jest.fn();
        const navigateBackSpy = jest.fn();
        const setNewDataSpy = jest.fn();

        const wrapper = shallow(
            <MnemonicNotConfirmedComponent
                navigation = { { state: { params: { setNewData: setNewDataSpy } } } }
                screenProps = { { 
                    redirectToLoginScreen: redirectToLoginScreenSpy,
                    navigateBack: navigateBackSpy
                } } />
        );

        wrapper.find('TouchableOpacity').forEach(child => {
            child.simulate('Press');
        });

        expect(redirectToLoginScreenSpy.mock.calls.length).toBe(1);
        expect(navigateBackSpy.mock.calls.length).toBe(1);
        expect(setNewDataSpy.mock.calls.length).toBe(1);
    });
    
});
