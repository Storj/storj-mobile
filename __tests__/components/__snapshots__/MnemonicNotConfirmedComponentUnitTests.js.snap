// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MnemonicNotConfirmedComponent renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MnemonicNotConfirmedComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <View
        style={
          Object {
            "paddingHorizontal": 40,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 140,
              "marginTop": 60,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#384B65",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 54,
                "lineHeight": 66,
                "marginLeft": 30,
              }
            }
          >
            Secret phrase backup failed
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 500,
              "justifyContent": "center",
              "marginTop": 150,
            }
          }
        >
          <Image
            resizeMode="contain"
            source="Error.png"
            style={
              Object {
                "height": 500,
                "width": 750,
              }
            }
          />
        </View>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#2782ff",
                "borderColor": "#2794FF",
                "borderRadius": 12,
                "borderWidth": 3,
                "height": 100,
                "justifyContent": "center",
                "marginTop": 120,
                "width": 670,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Montserrat-Bold",
                  "fontSize": 28,
                }
              }
            >
              Back
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#2794FF",
                "borderRadius": 6,
                "borderWidth": 3,
                "height": 100,
                "justifyContent": "center",
                "marginTop": 40,
                "width": 670,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#2794FF",
                  "fontFamily": "Montserrat-SemiBold",
                  "fontSize": 32,
                }
              }
            >
              Skip
            </Text>
          </View>
        </TouchableOpacity>
      </View>,
      "style": Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "flexDirection": "row",
                "height": 140,
                "marginTop": 60,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#384B65",
                  "fontFamily": "Montserrat-Bold",
                  "fontSize": 54,
                  "lineHeight": 66,
                  "marginLeft": 30,
                }
              }
            >
              Secret phrase backup failed
            </Text>
          </View>,
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 500,
                "justifyContent": "center",
                "marginTop": 150,
              }
            }
          >
            <Image
              resizeMode="contain"
              source="Error.png"
              style={
                Object {
                  "height": 500,
                  "width": 750,
                }
              }
            />
          </View>,
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#2782ff",
                  "borderColor": "#2794FF",
                  "borderRadius": 12,
                  "borderWidth": 3,
                  "height": 100,
                  "justifyContent": "center",
                  "marginTop": 120,
                  "width": 670,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 28,
                  }
                }
              >
                Back
              </Text>
            </View>
          </TouchableOpacity>,
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#2794FF",
                  "borderRadius": 6,
                  "borderWidth": 3,
                  "height": 100,
                  "justifyContent": "center",
                  "marginTop": 40,
                  "width": 670,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#2794FF",
                    "fontFamily": "Montserrat-SemiBold",
                    "fontSize": 32,
                  }
                }
              >
                Skip
              </Text>
            </View>
          </TouchableOpacity>,
        ],
        "style": Object {
          "paddingHorizontal": 40,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#384B65",
                  "fontFamily": "Montserrat-Bold",
                  "fontSize": 54,
                  "lineHeight": 66,
                  "marginLeft": 30,
                }
              }
            >
              Secret phrase backup failed
            </Text>,
            "style": Object {
              "flexDirection": "row",
              "height": 140,
              "marginTop": 60,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Secret phrase backup failed",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#384B65",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 54,
                "lineHeight": 66,
                "marginLeft": 30,
              },
            },
            "ref": null,
            "rendered": "Secret phrase backup failed",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Image
              resizeMode="contain"
              source="Error.png"
              style={
                Object {
                  "height": 500,
                  "width": 750,
                }
              }
            />,
            "style": Object {
              "alignItems": "center",
              "height": 500,
              "justifyContent": "center",
              "marginTop": 150,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "resizeMode": "contain",
              "source": "Error.png",
              "style": Object {
                "height": 500,
                "width": 750,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#2782ff",
                  "borderColor": "#2794FF",
                  "borderRadius": 12,
                  "borderWidth": 3,
                  "height": 100,
                  "justifyContent": "center",
                  "marginTop": 120,
                  "width": 670,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 28,
                  }
                }
              >
                Back
              </Text>
            </View>,
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 28,
                  }
                }
              >
                Back
              </Text>,
              "style": Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#2782ff",
                "borderColor": "#2794FF",
                "borderRadius": 12,
                "borderWidth": 3,
                "height": 100,
                "justifyContent": "center",
                "marginTop": 120,
                "width": 670,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Back",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "white",
                  "fontFamily": "Montserrat-Bold",
                  "fontSize": 28,
                },
              },
              "ref": null,
              "rendered": "Back",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#2794FF",
                  "borderRadius": 6,
                  "borderWidth": 3,
                  "height": 100,
                  "justifyContent": "center",
                  "marginTop": 40,
                  "width": 670,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#2794FF",
                    "fontFamily": "Montserrat-SemiBold",
                    "fontSize": 32,
                  }
                }
              >
                Skip
              </Text>
            </View>,
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#2794FF",
                    "fontFamily": "Montserrat-SemiBold",
                    "fontSize": 32,
                  }
                }
              >
                Skip
              </Text>,
              "style": Object {
                "alignItems": "center",
                "borderColor": "#2794FF",
                "borderRadius": 6,
                "borderWidth": 3,
                "height": 100,
                "justifyContent": "center",
                "marginTop": 40,
                "width": 670,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Skip",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#2794FF",
                  "fontFamily": "Montserrat-SemiBold",
                  "fontSize": 32,
                },
              },
              "ref": null,
              "rendered": "Skip",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <View
          style={
            Object {
              "paddingHorizontal": 40,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "height": 140,
                "marginTop": 60,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#384B65",
                  "fontFamily": "Montserrat-Bold",
                  "fontSize": 54,
                  "lineHeight": 66,
                  "marginLeft": 30,
                }
              }
            >
              Secret phrase backup failed
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 500,
                "justifyContent": "center",
                "marginTop": 150,
              }
            }
          >
            <Image
              resizeMode="contain"
              source="Error.png"
              style={
                Object {
                  "height": 500,
                  "width": 750,
                }
              }
            />
          </View>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#2782ff",
                  "borderColor": "#2794FF",
                  "borderRadius": 12,
                  "borderWidth": 3,
                  "height": 100,
                  "justifyContent": "center",
                  "marginTop": 120,
                  "width": 670,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 28,
                  }
                }
              >
                Back
              </Text>
            </View>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#2794FF",
                  "borderRadius": 6,
                  "borderWidth": 3,
                  "height": 100,
                  "justifyContent": "center",
                  "marginTop": 40,
                  "width": 670,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#2794FF",
                    "fontFamily": "Montserrat-SemiBold",
                    "fontSize": 32,
                  }
                }
              >
                Skip
              </Text>
            </View>
          </TouchableOpacity>
        </View>,
        "style": Object {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "height": 140,
                  "marginTop": 60,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#384B65",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 54,
                    "lineHeight": 66,
                    "marginLeft": 30,
                  }
                }
              >
                Secret phrase backup failed
              </Text>
            </View>,
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 500,
                  "justifyContent": "center",
                  "marginTop": 150,
                }
              }
            >
              <Image
                resizeMode="contain"
                source="Error.png"
                style={
                  Object {
                    "height": 500,
                    "width": 750,
                  }
                }
              />
            </View>,
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "backgroundColor": "#2782ff",
                    "borderColor": "#2794FF",
                    "borderRadius": 12,
                    "borderWidth": 3,
                    "height": 100,
                    "justifyContent": "center",
                    "marginTop": 120,
                    "width": 670,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Montserrat-Bold",
                      "fontSize": 28,
                    }
                  }
                >
                  Back
                </Text>
              </View>
            </TouchableOpacity>,
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderColor": "#2794FF",
                    "borderRadius": 6,
                    "borderWidth": 3,
                    "height": 100,
                    "justifyContent": "center",
                    "marginTop": 40,
                    "width": 670,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#2794FF",
                      "fontFamily": "Montserrat-SemiBold",
                      "fontSize": 32,
                    }
                  }
                >
                  Skip
                </Text>
              </View>
            </TouchableOpacity>,
          ],
          "style": Object {
            "paddingHorizontal": 40,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#384B65",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 54,
                    "lineHeight": 66,
                    "marginLeft": 30,
                  }
                }
              >
                Secret phrase backup failed
              </Text>,
              "style": Object {
                "flexDirection": "row",
                "height": 140,
                "marginTop": 60,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Secret phrase backup failed",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#384B65",
                  "fontFamily": "Montserrat-Bold",
                  "fontSize": 54,
                  "lineHeight": 66,
                  "marginLeft": 30,
                },
              },
              "ref": null,
              "rendered": "Secret phrase backup failed",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Image
                resizeMode="contain"
                source="Error.png"
                style={
                  Object {
                    "height": 500,
                    "width": 750,
                  }
                }
              />,
              "style": Object {
                "alignItems": "center",
                "height": 500,
                "justifyContent": "center",
                "marginTop": 150,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "resizeMode": "contain",
                "source": "Error.png",
                "style": Object {
                  "height": 500,
                  "width": 750,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "backgroundColor": "#2782ff",
                    "borderColor": "#2794FF",
                    "borderRadius": 12,
                    "borderWidth": 3,
                    "height": 100,
                    "justifyContent": "center",
                    "marginTop": 120,
                    "width": 670,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Montserrat-Bold",
                      "fontSize": 28,
                    }
                  }
                >
                  Back
                </Text>
              </View>,
              "onPress": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Montserrat-Bold",
                      "fontSize": 28,
                    }
                  }
                >
                  Back
                </Text>,
                "style": Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#2782ff",
                  "borderColor": "#2794FF",
                  "borderRadius": 12,
                  "borderWidth": 3,
                  "height": 100,
                  "justifyContent": "center",
                  "marginTop": 120,
                  "width": 670,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Back",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "white",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 28,
                  },
                },
                "ref": null,
                "rendered": "Back",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderColor": "#2794FF",
                    "borderRadius": 6,
                    "borderWidth": 3,
                    "height": 100,
                    "justifyContent": "center",
                    "marginTop": 40,
                    "width": 670,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#2794FF",
                      "fontFamily": "Montserrat-SemiBold",
                      "fontSize": 32,
                    }
                  }
                >
                  Skip
                </Text>
              </View>,
              "onPress": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#2794FF",
                      "fontFamily": "Montserrat-SemiBold",
                      "fontSize": 32,
                    }
                  }
                >
                  Skip
                </Text>,
                "style": Object {
                  "alignItems": "center",
                  "borderColor": "#2794FF",
                  "borderRadius": 6,
                  "borderWidth": 3,
                  "height": 100,
                  "justifyContent": "center",
                  "marginTop": 40,
                  "width": 670,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Skip",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "#2794FF",
                    "fontFamily": "Montserrat-SemiBold",
                    "fontSize": 32,
                  },
                },
                "ref": null,
                "rendered": "Skip",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
